### PR TITLE
feat(Azure): Add 4 new checks related to SQLServer and Vulnerability Assessment

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

Add new checks and related tests:

- [x] `sqlserver_microsoft_defender_enabled`
- [x] `sqlserver_va_periodic_recurring_scans_enabled`
- [x] `sqlserver_va_scan_reports_configured`
- [x] `sqlserver_va_emails_notifications_admins_enabled`

### Description

The task that are made for each check:

- Ensure that Microsoft Defender for SQL is set to 'On' for critical SQL Servers
- Ensure that Vulnerability Assessment (VA) setting 'Periodic recurring scans' is set to 'on' for each SQL server
- Ensure that Vulnerability Assessment (VA) setting 'Send scan reports to' is configured for a SQL server
- Ensure that Vulnerability Assessment (VA) setting 'Also send email notifications to admins and subscription owners' is set for each SQL Server

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "ResourceType": "SQLServer",
+  "Description": "Ensure that Microsoft Defender for SQL is set to 'On' for critical SQL Servers",
+  "Risk": "Microsoft Defender for SQL is a unified package for advanced SQL security capabilities. Microsoft Defender is available for Azure SQL Database, Azure SQL Managed  classifying sensitive data, surfacing and mitigating potential database vulnerabilities, and detecting anomalous activities that could indicate a threat to your database. It provides a single go-to location for enabling and managing these capabilities.",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/azure-sql/database/azure-defender-for-sql?view=azuresql",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/azure/SecurityCenter/defender-azure-sql.html",
+      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-azure-defender-is-set-to-on-for-azure-sql-database-servers"
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.metadata.json]:
```suggestion
      "Terraform": ""
```

File: prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        for subscription, sql_servers in sqlserver_client.sql_servers.items():
+            for sql_server in sql_servers:
+                if sql_server.security_alert_policies is not None:
+                    report = Check_Report_Azure(self.metadata())
+                    report.subscription = subscription
+                    report.resource_name = sql_server.name
+                    report.resource_id = sql_server.id
+                    report.status = "FAIL"
+                    report.status_extended = f"SQL Server {sql_server.name} from subscription {subscription} has microsoft defender disabled."
+                    if sql_server.security_alert_policies.state == "Enabled":
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py:17]: is this the same as `Ensure that ThreatDetectionState is set to Enabled.` ?

File: prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+from prowler.lib.check.models import Check, Check_Report_Azure
+from prowler.providers.azure.services.sqlserver.sqlserver_client import sqlserver_client
+
+
+class sqlserver_microsoft_defender_enabled(Check):
+    def execute(self) -> Check_Report_Azure:
+        findings = []
+        for subscription, sql_servers in sqlserver_client.sql_servers.items():
+            for sql_server in sql_servers:
+                if sql_server.security_alert_policies is not None:
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py]:
```suggestion
                if sql_server.security_alert_policies:
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "ResourceType": "SQLServer",
+  "Description": "Enable Vulnerability Assessment (VA) setting 'Also send email notifications to admins and subscription owners'.",
+  "Risk": "VA scan reports and alerts will be sent to admins and subscription owners by enabling setting 'Also send email notifications to admins and subscription owners'. This may help in reducing time required for identifying risks and taking corrective measures.",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://www.tenable.com/audits/items/CIS_Microsoft_Azure_Foundations_L2_v1.3.1.audit:c4e2d4019a12b8410b55a6cf2d838d7b",
+      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-va-setting-also-send-email-notifications-to-admins-and-subscription-owners-is-set-for-an-sql-server"
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json]:
```suggestion
      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-va-setting-also-send-email-notifications-to-admins-and-subscription-owners-is-set-for-an-sql-server#terraform"
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "medium",
+  "ResourceType": "SQLServer",
+  "Description": "Enable Vulnerability Assessment (VA) setting 'Also send email notifications to admins and subscription owners'.",
+  "Risk": "VA scan reports and alerts will be sent to admins and subscription owners by enabling setting 'Also send email notifications to admins and subscription owners'. This may help in reducing time required for identifying risks and taking corrective measures.",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://www.tenable.com/audits/items/CIS_Microsoft_Azure_Foundations_L2_v1.3.1.audit:c4e2d4019a12b8410b55a6cf2d838d7b",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json]:
```suggestion
      "Other": "",
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.resource_name = sql_server.name
+                report.resource_id = sql_server.id
+                report.status = "FAIL"
+                report.status_extended = f"SQL Server {sql_server.name} from subscription {subscription} has vulnerability assessment disabled."
+                if (
+                    sql_server.vulnerability_assessment
+                    and sql_server.vulnerability_assessment.storage_container_path
+                    is not None
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py]:
```suggestion
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+                ):
+                    report.status_extended = f"SQL Server {sql_server.name} from subscription {subscription} has vulnerability assessment enabled but no scan reports configured for subscription admins."
+                    if (
+                        sql_server.vulnerability_assessment.recurring_scans.email_subscription_admins
+                        is not None
+                        and sql_server.vulnerability_assessment.recurring_scans.email_subscription_admins
+                    ):
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py]:
```suggestion
                    if (
                        sql_server.vulnerability_assessment.recurring_scans and sql_server.vulnerability_assessment.recurring_scans.email_subscription_admins
                    ):
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.metadata.json [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  "ResourceType": "SQLServer",
+  "Description": "Enable Vulnerability Assessment (VA) Periodic recurring scans for critical SQL servers and corresponding SQL databases.",
+  "Risk": "VA setting 'Periodic recurring scans' schedules periodic (weekly) vulnerability scanning for the SQL server and corresponding Databases. Periodic and regular vulnerability scanning provides risk visibility based on updated known vulnerability signatures and best practices.",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/azure/Sql/periodic-vulnerability-scans.html#",
+      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-va-setting-periodic-recurring-scans-is-enabled-on-a-sql-server"
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.metadata.json]:
```suggestion
      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-va-setting-periodic-recurring-scans-is-enabled-on-a-sql-server#terraform"
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.resource_name = sql_server.name
+                report.resource_id = sql_server.id
+                report.status = "FAIL"
+                report.status_extended = f"SQL Server {sql_server.name} from subscription {subscription} has vulnerability assessment disabled."
+                if (
+                    sql_server.vulnerability_assessment
+                    and sql_server.vulnerability_assessment.storage_container_path
+                    is not None
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py]:
```suggestion
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+                ):
+                    report.status_extended = f"SQL Server {sql_server.name} from subscription {subscription} has vulnerability assessment enabled but no recurring scans."
+                    if sql_server.vulnerability_assessment.recurring_scans.is_enabled:
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py]:
```suggestion
                    if sql_server.vulnerability_assessment.recurring_scans and sql_server.vulnerability_assessment.recurring_scans.is_enabled:
```

File: prowler/providers/azure/services/sqlserver/sqlserver_service.py [hunk 11]
```diff
@@ -44,6 +45,11 @@ def __get_sql_servers__(self):
                     vulnerability_assessment = self.__get_vulnerability_assesments__(
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_service.py:42]: Please, add separated functions for `list_by_server`?

File: prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  "ResourceType": "SQLServer",
+  "Description": "Configure 'Send scan reports to' with email addresses of concerned data owners/stakeholders for a critical SQL servers.",
+  "Risk": "Vulnerability Assessment (VA) scan reports and alerts will be sent to email addresses configured at 'Send scan reports to'. This may help in reducing time required for identifying risks and taking corrective measures",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://www.tenable.com/audits/items/CIS_Microsoft_Azure_Foundations_L2_v1.3.1.audit:7a3f11f12c52122fc23badc7c808ebe9",
+      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-va-setting-send-scan-reports-to-is-configured-for-a-sql-server"
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json]:
```suggestion
      "Terraform": "https://docs.bridgecrew.io/docs/ensure-that-va-setting-send-scan-reports-to-is-configured-for-a-sql-server#terraform"
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "medium",
+  "ResourceType": "SQLServer",
+  "Description": "Configure 'Send scan reports to' with email addresses of concerned data owners/stakeholders for a critical SQL servers.",
+  "Risk": "Vulnerability Assessment (VA) scan reports and alerts will be sent to email addresses configured at 'Send scan reports to'. This may help in reducing time required for identifying risks and taking corrective measures",
+  "RelatedUrl": "https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "https://www.tenable.com/audits/items/CIS_Microsoft_Azure_Foundations_L2_v1.3.1.audit:7a3f11f12c52122fc23badc7c808ebe9",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json]:
```suggestion
      "Other": "",
```

File: prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.py [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+                        sql_server.vulnerability_assessment.recurring_scans.email_subscription_admins
+                        is not None
+                        and sql_server.vulnerability_assessment.recurring_scans.email_subscription_admins
+                    ) or (
+                        sql_server.vulnerability_assessment.recurring_scans.emails
+                        is not None
+                        and len(
+                            sql_server.vulnerability_assessment.recurring_scans.emails
+                        )
+                        > 0
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.py]:
```suggestion
                    if sql_server.vulnerability_assessment.recurring_scans and (
                        sql_server.vulnerability_assessment.recurring_scans.email_subscription_admins or 
                        (sql_server.vulnerability_assessment.recurring_scans.emails
                        and len(
                            sql_server.vulnerability_assessment.recurring_scans.emails
                        )
                        > 0
```

File: prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.metadata.json (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.metadata.json]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py:17]: Yes

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_microsoft_defender_enabled/sqlserver_microsoft_defender_enabled.py]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json (hunk 5 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.metadata.json]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py (hunk 6 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py (hunk 7 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_emails_notifications_admins_enabled/sqlserver_va_emails_notifications_admins_enabled.py]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.metadata.json (hunk 8 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.metadata.json]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py (hunk 9 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py (hunk 10 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_periodic_recurring_scans_enabled/sqlserver_va_periodic_recurring_scans_enabled.py]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json (hunk 12 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json (hunk 13 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.metadata.json]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.py (hunk 14 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_va_scan_reports_configured/sqlserver_va_scan_reports_configured.py]: Done!

File: prowler/providers/azure/services/sqlserver/sqlserver_service.py (hunk 11 above)
Comment (pedrooot) [reply prowler/providers/azure/services/sqlserver/sqlserver_service.py:42]: Good idea! Done

**Review by MrCloudSec — APPROVED** [review]
LGTM !!

**Pull request merged**